src/core/tree_methods.ts

```diff
@@ -0,0 +1,76 @@
+export class TreeNode {
+  readonly content: string;
+  readonly is_done: boolean;
+  readonly children: TreeNode[];
```

Comment:
Nit: whitespace is a thing

Reply:
?

Reply:
Logical blocks of code should be separated by whitespace - so the layout should be more like:
```
[fields]

[constructor(s)]

[method0]

[method1]
```

Rather than:
```[fields];[constructors];[...];``` - which is valid TS

Ofc I'm exaggerating, but you get the point ;)

Reply:
lol so you mean "add 'enter' after fields".